tests/compiler/compiler.vitest.test.ts

```diff
@@ -2,22 +2,22 @@ import path from 'path';
 import { describe, beforeAll, afterAll, test, expect } from 'vitest';
 import { createCompiler, normalizePath } from '@vanilla-extract/integration';
 
-function getLocalFiles(files: Set<string>) {
-  const posixDirname = normalizePath(__dirname);
-
-  return [...files]
-    .map(normalizePath)
-    .filter((file) => file.startsWith(posixDirname))
-    .map((file) => file.replace(posixDirname, ''));
-}
+// Vitest has trouble with snapshots that don't contain wrapping quotes. See this regex and the functions above/below it:
+// https://github.com/vitest-dev/vitest/blob/ae73f2737607a878ba589d548aa6f8ba639dc07c/packages/snapshot/src/port/inlineSnapshot.ts#L96
+// We want to replace __dirname in strings, so we need to add a snapshot serializer for strings.
+// But we also need to add quotes around the string to make sure Vitest doesn't get confused.
+// In fact, just updating the snapshot from now on will mangle the test file ¯\_(ツ)_/¯
+expect.addSnapshotSerializer({
+  test: (val) => typeof val === 'string',
+  print: (val) =>
+    '"' +
+    (val as string).replaceAll(normalizePath(__dirname), '{{__dirname}}') +
+    '"',
+});
 
-describe('compiler', () => {
+describe('compiler running in Vitest', () => {
```

Comment:
Assuming this test file is just to ensure compatibility with vitest, then IMO it's worth leaving a comment here to tell future contributors to add tests to `compiler.test.ts` file going forward.